### PR TITLE
Hyundai CAN: Explicitly parse gear shifter values for `EV, HEV, PHEV`

### PR DESCRIPTION
**Pull request by sunnyhaibin**
**Description**
When the car is In manumatic or different drive modes (e.g. Sport, Snow), openpilot currently does not recognize the signals and returns with "Gear not D" event and exits controls.

**Solution**
This PR adds support to parse different forward gear signals and adds additional undefined forward gear signals in the DBC.

**Route ID**
- `e1107f9d04dfb1e2|2024-02-16--14-51-43`
- `26a38e4bf36bb74a|2023-11-15--10-11-14`

**Requires**
- https://github.com/commaai/opendbc/pull/981

### CODE REVIEW

File: selfdrive/car/hyundai/carstate.py [hunk 1]
```diff
@@ -29,6 +29,8 @@ def __init__(self, CP):
                           "GEAR_SHIFTER"
     if CP.carFingerprint in CANFD_CAR:
       self.shifter_values = can_define.dv[self.gear_msg_canfd]["GEAR"]
+    elif self.CP.carFingerprint in (HYBRID_CAR | EV_CAR):
+      self.shifter_values = can_define.dv["ELECT_GEAR"]["Elect_Gear_Shifter"]
```
Comment (sshane) [inline selfdrive/car/hyundai/carstate.py:33]: This just makes this gear `GearShifter.sport` from `GearShifter.unknown`, `wrongGear` alert still checks if it's drive. You need to add an extra gear to `create_common_events`

File: selfdrive/car/hyundai/carstate.py [hunk 2]
```diff
@@ -29,6 +29,8 @@ def __init__(self, CP):
...
     elif self.CP.carFingerprint in CAN_GEARS["use_cluster_gears"]:
       self.shifter_values = can_define.dv["CLU15"]["CF_Clu_Gear"]
     elif self.CP.carFingerprint in CAN_GEARS["use_tcu_gears"]:
```
Comment (sshane) [inline selfdrive/car/hyundai/carstate.py:36]: remove `# preferred and elect gear methods use same definition`comment below

File: selfdrive/car/hyundai/carstate.py (hunk 1 above)
Comment (sunnyhaibin) [reply selfdrive/car/hyundai/carstate.py:33]: @sshane Updated.

File: selfdrive/car/hyundai/carstate.py (hunk 2 above)
Comment (sunnyhaibin) [reply selfdrive/car/hyundai/carstate.py:36]: @sshane Updated.

**Comment by adeebshihadeh** [comment]
Now that the big car dataset is out, you should be able to verify this the same way we would do. There's some example notebooks in tools/car_porting. If you have any questions about the process, let's discuss in #dev-openpilot-private.

**Comment by sshane** [comment]
We've moved the car interfacing code to our [`opendbc`](https://github.com/commaai/opendbc) repository, which is now the new home for car ports and fingerprints. Please re-open your pull request against [`opendbc`](https://github.com/commaai/opendbc) at your convenience by using this command below. This will transform all changes under `selfdrive/car/` to `opendbc_repo/opendbc/car/`. Make sure you have initialized submodules beforehand and have checked out the latest `opendbc` commit.

```bash
PR_NUMBER=33045
curl -L https://github.com/commaai/openpilot/pull/$PR_NUMBER.patch | sed -e 's/selfdrive\/car/opendbc_repo\/opendbc\/car/g' | git apply -v --reject
```

Simply replace the PR number with your own. Once done, add the files, fix any conflicts, and open a new PR. Alternatively, you may start a new PR from scratch if that is easier for you.

**Pull request closed without merging**